plugins/ethropsten_swap_plugin/include/eosio/ethropsten_swap_plugin/ethropsten_swap_plugin.hpp

```diff
@@ -0,0 +1,159 @@
+#include <websocketpp/client.hpp>
+#include <websocketpp/common/thread.hpp>
+
+#define FC_LOG_WAIT_AND_CONTINUE(...)  \
+    catch( const boost::interprocess::bad_alloc& ) {\
+       throw;\
+    } catch( fc::exception& er ) { \
+       wlog( "${details}", ("details",er.to_detail_string()) ); \
+       sleep(wait_for_eth_node); \
+       continue; \
+    } catch( const std::exception& e ) {  \
+       fc::exception fce( \
+                 FC_LOG_MESSAGE( warn, "rethrow ${what}: ",FC_FORMAT_ARG_PARAMS( __VA_ARGS__  )("what",e.what()) ), \
+                 fc::std_exception_code,\
+                 BOOST_CORE_TYPEID(e).name(), \
+                 e.what() ) ; \
+       wlog( "${details}", ("details",fce.to_detail_string()) ); \
+       sleep(wait_for_eth_node); \
+       continue; \
+    } catch( ... ) {  \
+       fc::unhandled_exception e( \
+                 FC_LOG_MESSAGE( warn, "rethrow", FC_FORMAT_ARG_PARAMS( __VA_ARGS__) ), \
+                 std::current_exception() ); \
+       wlog( "${details}", ("details",e.to_detail_string()) ); \
+       sleep(wait_for_eth_node); \
+       continue; \
+    }
+
+#define FC_LOG_AND_RETURN(...) \
+    catch( const boost::interprocess::bad_alloc& ) {\
+       throw;\
+    } catch( fc::exception& er ) { \
+       wlog( "${details}", ("details",er.to_detail_string()) ); \
+       return; \
+    } catch( const std::exception& e ) {  \
+       fc::exception fce( \
+                 FC_LOG_MESSAGE( warn, "rethrow ${what}: ",FC_FORMAT_ARG_PARAMS( __VA_ARGS__  )("what",e.what()) ), \
+                 fc::std_exception_code,\
+                 BOOST_CORE_TYPEID(e).name(), \
+                 e.what() ) ; \
+       wlog( "${details}", ("details",fce.to_detail_string()) ); \
+       return; \
+    } catch( ... ) {  \
+       fc::unhandled_exception e( \
+                 FC_LOG_MESSAGE( warn, "rethrow", FC_FORMAT_ARG_PARAMS( __VA_ARGS__) ), \
+                 std::current_exception() ); \
+       wlog( "${details}", ("details",e.to_detail_string()) ); \
+       return; \
+    }
+
+typedef websocketpp::client<websocketpp::config::asio_tls_client> client;
+typedef websocketpp::lib::lock_guard<websocketpp::lib::mutex> scoped_lock;
+typedef websocketpp::config::asio_client::message_type::ptr message_ptr;
+namespace eosio {
+
+using namespace appbase;
+
+constexpr int32_t block_interval_ms = 500;
+constexpr int64_t block_timestamp_epoch = 946684800000ll;  // epoch is year 2000
+typedef eosio::chain::block_timestamp<block_interval_ms, block_timestamp_epoch> epoch_block_timestamp;
+
+const char* rem_token_id              = "REM";
+const char* eth_swap_contract_address = "0x9fB8A18fF402680b47387AE0F4e38229EC64f098";
+const char* eth_swap_request_event    = "0x0e918020302bf93eb479360905c1535ba1dbc8aeb6d20eff433206bf4c514e13";
+const char* return_chain_id           = "ethropsten";
+const char* chain_id                  = "93ece941df27a5787a405383a66a7c26d04e80182adf504365710331ac0625a7";
```

Comment:
let's make it possible to override these params in the config

Reply:
Done.